### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by davidtheclark**
Addresses #1310.

As you can see from commits, I've tried to:

- Remove emptyLineBefore option for declaration-block-properties-order
- Remove hierarchicalSelectors option from indentation
- Remove stylelint.utils.cssWordIsVariable
- Remove support for non-namespaced plugin rules
- Remove deprecated options from font-family-name-quotes
- Remove deprecated options of function-url-quotes (this plus the above = "contextual quote rule options")

I believe that addresses all the checkboxes in #1310.

Of course the big points to help check here are:
- Documentation is updated accurately and is still helpful.
- No still important tests were removed.

### CODE REVIEW

File: src/__tests__/plugins-test.js [hunk 1]
```diff
@@ -171,27 +171,6 @@ test("module providing an array of plugins", t => {
   t.plan(planned)
 })
 
-test("deprecation warning for slashless plugin rule names", t => {
```
Comment (m-allanson) [inline src/__tests__/plugins-test.js]: Should this test be replaced with a test to check for a config error? Or should the file `plugin-slashless-warn-about-foo.js` be removed?

Comment (davidtheclark) [inline src/__tests__/plugins-test.js]: Yeah, these are good ideas. Thanks for looking closely!

Comment (davidtheclark) [inline src/__tests__/plugins-test.js]: In latest commit added a test, as you can see below.

**Comment by jeddy3** [comment]
Oh wow. Great to see this. I'm confident we're on a more sustainable trajectory with these removals.

Thanks also for bringing the contextual quote rules inline with the `expected` and `rejected` convention.

Feel free to merge after a rebase.

**Comment by ntwb** [comment]
Nice, and a nice bump in coverage also 👍 
> Coverage increased (+1.2%) to 98.255%

**Pull request merged**